internal/commands/vaultsecrets/secrets/create_test.go

```diff
@@ -0,0 +1,227 @@
+	mock_secret_service "github.com/hashicorp/hcp/internal/pkg/api/mocks/github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCmdCreate(t *testing.T) {
+	t.Parallel()
+
+	testProfile := func(t *testing.T) *profile.Profile {
+		tp := profile.TestProfile(t).SetOrgID("123").SetProjectID("456")
+		tp.VaultSecrets = &profile.VaultSecretsConf{
+			AppName: "test-app-name",
+		}
+		return tp
+	}
+
+	cases := []struct {
+		Name    string
+		Args    []string
+		Profile func(t *testing.T) *profile.Profile
+		Error   string
+		Expect  *CreateOpts
+	}{
+		{
+			Name:    "Failed: No secret name arg specified",
+			Profile: testProfile,
+			Args:    []string{},
+			Error:   "ERROR: accepts 1 arg(s), received 0",
+		},
+		{
+			Name:    "Good: Secret name arg specified",
+			Profile: testProfile,
+			Args:    []string{"test"},
+			Expect: &CreateOpts{
+				AppName:    testProfile(t).VaultSecrets.AppName,
+				SecretName: "test",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			t.Parallel()
+			r := require.New(t)
+
+			// Create a context.
+			io := iostreams.Test()
+			ctx := &cmd.Context{
+				IO:          io,
+				Profile:     c.Profile(t),
+				Output:      format.New(io),
+				HCP:         &client.Runtime{},
+				ShutdownCtx: context.Background(),
+			}
+
+			var gotOpts *CreateOpts
+			createCmd := NewCmdCreate(ctx, func(o *CreateOpts) error {
+				gotOpts = o
+				gotOpts.AppName = "test-app-name"
+				return nil
+			})
+			createCmd.SetIO(io)
+
+			code := createCmd.Run(c.Args)
+			if c.Error != "" {
+				r.NotZero(code)
+				r.Contains(io.Error.String(), c.Error)
+				return
+			}
+
+			r.Zero(code, io.Error.String())
+			r.NotNil(gotOpts)
+			r.Equal(c.Expect.AppName, gotOpts.AppName)
+			r.Equal(c.Expect.SecretName, gotOpts.SecretName)
+		})
+	}
+}
+
+func TestCreateRun(t *testing.T) {
+	t.Parallel()
+
+	testProfile := func(t *testing.T) *profile.Profile {
+		tp := profile.TestProfile(t).SetOrgID("123").SetProjectID("456")
+		tp.VaultSecrets = &profile.VaultSecretsConf{
+			AppName: "test-app-name",
+		}
+		return tp
+	}
+	testSecretValue := "my super secret value"
+
+	cases := []struct {
+		Name             string
+		RespErr          bool
+		ReadViaStdin     bool
+		EmptySecretValue bool
+		ErrMsg           string
+		MockCalled       bool
+		AugmentOpts      func(*CreateOpts)
+	}{
+		{
+			Name:   "Failed: Read via stdin as hypen not supplied for --data-file flag",
+			ErrMsg: "data file path is required",
+		},
+		{
+			Name:             "Failed: Read empty secret value via stdin",
+			EmptySecretValue: true,
+			ReadViaStdin:     true,
+			RespErr:          true,
```

Comment:
nit: Do we need `RespErr` and `ErrMsg` since we can just use `ErrMsg` to check if an error is expected?

Reply:
mostly following convention through this repo. If you search for RespErr, you will see it being used as a bool.